pkg/apis/migration/v1alpha1/migplan_types.go

```diff
@@ -371,36 +372,68 @@ func (r *MigPlan) BuildRegistryDC(storage *MigStorage, name, dirName string) (*a
 			Namespace: VeleroNamespace,
 		},
 	}
-	err := r.UpdateRegistryDC(storage, deploymentconfig, name, dirName)
+	err := r.UpdateRegistryDC(storage, deploymentconfig, proxySecret, name, dirName)
 	return deploymentconfig, err
 }
 
-func buildProxyEnvVars() []kapi.EnvVar {
-	envVars := []kapi.EnvVar{}
-	if found, s := Settings.HasProxyVar(settings.HttpProxy); found {
-		envVars = append(envVars, kapi.EnvVar{
-			Name:  settings.HttpProxy,
-			Value: s,
-		})
+// Get registry proxy secret for registry DC
+// Returns nil if secret isn't found or no configuration exists
+func (r *MigPlan) GetRegistryProxySecret(client k8sclient.Client) (*kapi.Secret, error) {
+	list := kapi.SecretList{}
+	selector := k8sLabels.SelectorFromSet(map[string]string{
+		"migration-proxy-config": "true",
+	})
+	err := client.List(
+		context.TODO(),
+		&k8sclient.ListOptions{
+			Namespace:     VeleroNamespace,
+			LabelSelector: selector,
+		},
+		&list,
+	)
+	// If error listing secrets return no env vars
+	if err != nil {
+		return nil, err
 	}
-	if found, s := Settings.HasProxyVar(settings.HttpsProxy); found {
-		envVars = append(envVars, kapi.EnvVar{
-			Name:  settings.HttpsProxy,
-			Value: s,
-		})
+	// If no secrets found, return nil and no error
+	// This means no proxy is configured
+	if len(list.Items) == 0 {
+		return nil, nil
 	}
-	if found, s := Settings.HasProxyVar(settings.NoProxy); found {
-		envVars = append(envVars, kapi.EnvVar{
-			Name:  settings.NoProxy,
-			Value: s,
-		})
+	if len(list.Items) > 1 {
+		return nil, errors.New("found multiple proxy config secrets")
+	}
+	secret, err := GetSecret(
```

Comment:
Why fetch the secret when it's already in list.Items[0]?

Reply:
I don't know :)